pkg/odo/cli/component/exec.go

```diff
@@ -0,0 +1,107 @@
+package component
+
+import (
+	"fmt"
+	projectCmd "github.com/openshift/odo/pkg/odo/cli/project"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	odoutil "github.com/openshift/odo/pkg/odo/util"
+	"github.com/openshift/odo/pkg/odo/util/completion"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/odo/util/pushtarget"
+	"github.com/openshift/odo/pkg/util"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+	"path/filepath"
+)
+
+// ExecRecommendedCommandName is the recommended exec command name
+const ExecRecommendedCommandName = "exec"
+
+var execExample = ktemplates.Examples(`  # Executes a command inside the component
+%[1]s
+`)
+
+// ExecOptions contains exec options
+type ExecOptions struct {
+	componentContext string
+	componentOptions *ComponentOptions
+	devfilePath      string
+	namespace        string
+
+	command []string
+}
+
+// NewExecOptions returns new instance of ExecOptions
+func NewExecOptions() *ExecOptions {
+	return &ExecOptions{
+		componentOptions: &ComponentOptions{},
+	}
+}
+
+// Complete completes exec args
+func (eo *ExecOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	if cmd.ArgsLenAtDash() <= -1 {
+		return fmt.Errorf("no command was given for the exec command")
```

Comment:
could we elaborate error message here
`command not provided`
` please provide a command to execute, odo exec -- <command to be executed> `